.github/workflows/publish.yml

```diff
@@ -0,0 +1,26 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build Online Document
+      run: |
+        git config --local --get remote.origin.url
+        cd docs/build_docs
+        bash build.sh latest
+
+    - name: Push to github
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ./build_tmp/gh-pages
+        publish_branch: gh-pages
```

Comment:
We can use the official action for publishing to github pages

```
      - name: Upload documentation archive
        uses: actions/upload-pages-artifact@v3
        with:
          path: 'dist/html'
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v4
```

I also don't think the gh-pages branch is needed